### DISCUSSION TITLE
Is it normal to slightly exceed the boundary value？

### DISCUSSION THREAD

**Discussion by willi7758521** [osqp/osqp#557] [General]
Although with the non-negtive constrain,the solution of osqp is still have some negtive values, the range of them is about -10^(-2) to 0;
Is this phenomeno normal? Thanks

**Comment by goulart-paul** [comment]
Yes.   It will only satisfy constraints to within the termination criteria you set for the solver.  This means that when the solver stops you may well have small constraint violations still.

**Comment by tomwenseleers** [comment]
Would it not make sense to clip the returned coefficient estimates within the requested constraints?

**Comment by goulart-paul** [comment]
It would perhaps make sense to do that if the constraints were all simple upper/lower bounds, but it's not clear what should be done in the general case that a constraint is `l \le Ax \le u`.    We don't make any special check for the former condition, so just return whatever we get.

**Comment by wadasdw88s** [comment]
same question!
how to aovid the situation?
